embedded-redis/src/main/java/com/playtika/test/redis/EmbeddedRedisClusterBootstrapConfiguration.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.MountableFile;
+import redis.clients.jedis.Jedis;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.playtika.test.common.utils.ContainerUtils.containerLogsConsumer;
+import static com.playtika.test.redis.EnvUtils.registerRedisEnvironment;
+import static com.playtika.test.redis.RedisProperties.BEAN_NAME_EMBEDDED_REDIS;
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+@Configuration
+@AutoConfigureOrder(Ordered.HIGHEST_PRECEDENCE)
+@Conditional(EmbeddedRedisClusterBootstrapConfiguration.EmbeddedRedisClusterCondition.class)
+@EnableConfigurationProperties(RedisProperties.class)
+public class EmbeddedRedisClusterBootstrapConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    RedisStatusCheck redisStartupCheckStrategy() {
+        return new RedisStatusCheck();
+    }
+
+    @Bean(name = BEAN_NAME_EMBEDDED_REDIS, destroyMethod = "stop")
+    public GenericContainer redis(ConfigurableEnvironment environment,
+                                  RedisProperties properties,
+                                  RedisStatusCheck redisStatusCheck) {
+
+        log.info("Starting Redis cluster. Docker image: {}", properties.dockerImage);
+
+        // CLUSTER SLOTS command returns IP:port for each node, so ports outside and inside
+        // container must be the same
+        GenericContainer redis =
+                new FixedHostPortGenericContainer(properties.dockerImage)
+                        .withFixedExposedPort(properties.port, properties.port)
+                        .withLogConsumer(containerLogsConsumer(log))
+                        .withEnv("REDIS_USER", properties.getUser())
+                        .withEnv("REDIS_PASSWORD", properties.getPassword())
+                        .withCopyFileToContainer(MountableFile.forClasspathResource("redis-cluster.conf"), "/data/redis-cluster.conf")
+                        .withCopyFileToContainer(MountableFile.forClasspathResource("nodes.conf"), "/data/nodes.conf")
+                        .withCommand("redis-server", "/data/redis-cluster.conf", "--requirepass", properties.getPassword())
+                        .waitingFor(redisStatusCheck)
+                        .withStartupTimeout(properties.getTimeoutDuration());
+        redis.start();
+        MapPropertySource redisEnv = registerRedisEnvironment(redis, properties.port, environment, properties);
+        log.info("Started Redis cluster. Connection details: {}", redisEnv);
+        waitUntilClusterStateOk(properties);
+        return redis;
+    }
+
+    private void waitUntilClusterStateOk(RedisProperties properties) {
```

Comment:
you can provide custom wait strategy instead of checking it manually.
please check an example here:
https://github.com/Playtika/testcontainers-spring-boot/blob/develop/embedded-aerospike/src/main/java/com/playtika/test/aerospike/AerospikeWaitStrategy.java

Reply:
Thanks, done. Removed awaitility too